### PR TITLE
OCP Update variable filter to consider go_template

### PR DESCRIPTION
**Pull request by Vincent056**
#### Description:
Update the variable filter to find if a rule is using go-template, if so find any var being used, and add them to the var list for that rule, Compliance Operator use `[go template](https://pkg.go.dev/text/template)` to do additional processing, sometimes we reference a xccdf variable within templated content.

### CODE REVIEW

**Comment by Vincent056** [comment]
> /test e2e-aws-ocp4-pci-dss /test e2e-aws-ocp4-pci-dss-node

we have to merge https://github.com/ComplianceAsCode/content/pull/11906 before we can test

**Comment by xiaojiey** [comment]
@Vincent056 seems one issue to fix: https://codeclimate.com/github/ComplianceAsCode/content/pull/11906. Could you please take a look? Thanks.

**Comment by rhmdnd** [comment]
The stig e2e failure is unrelated and being addressed in a separate PR.

**Review by rhmdnd — APPROVED** [review]
/lgtm

Looks good from a OpenShift content perspective.

**Comment by rhmdnd** [comment]
@Mab879 does this seem reasonable from the build perspective?

**Comment by rhmdnd** [comment]
@Honny1 how do you feel able this approach given the recent refactors to the build system made in https://github.com/ComplianceAsCode/content/pull/11858

**Review by Honny1 — CHANGES_REQUESTED** [review]
I built and compared `ocp4` ds from this PR branch and the main branch and found no problem. However, I would like to ask you to create tests for this problem to avoid similar problems in future changes. Also, I would like to ask why you are processing the DS after the build? Wouldn't it be better to extend the macros during the build?

File: build-scripts/build_xccdf.py [hunk 1]
```diff
@@ -114,9 +114,20 @@ def get_linked_xccdf(loader, xccdftree, args):
     return oval_linker, xccdftree
 
 
+def get_variables_from_go_templating(rule, var_ids):
+    go_templating_pattern = re.compile(r"{{(.*?)}}")
+    go_templating_var_pattern = re.compile(r"\.([a-zA-Z0-9_]+)")
+    for ele in rule.itertext():
```
[the remark this answers is not in the document]
Comment (Vincent056) [reply build-scripts/build_xccdf.py:120]: they will only present in the text part of the rule

[the remark this answers is not in the document]
Comment (Vincent056) [reply build-scripts/build_xccdf.py:120]: I think fix is also part of the rule

**Comment by Vincent056** [comment]
> I built and compared `ocp4` ds from this PR branch and the main branch and found no problem. However, I would like to ask you to create tests for this problem to avoid similar problems in future changes. Also, I would like to ask why you are processing the DS after the build? Wouldn't it be better to extend the macros during the build?

Thanks for the review, we have e2e test, but we don't run that in every PR, for example you could run `/test 4.15-e2e-aws-ocp4-high` for one of the test, I can add the unit test in another PR since this is blocking our CI and release.

The go_template is use to in our operator, so that we can process the remediation to use xccdf variables as well as other part of rule to be more dynamic.

**Comment by Honny1** [comment]
@Vincent056 
Okay, please add a unit test in another PR. 

Well, I think the `sub` element would do the same magic.  See rule: `xccdf_org.ssgproject.content_rule_enable_fips_mode` 
Or is there some other reason for this approach?

**Comment by Vincent056** [comment]
> xccdf_org.ssgproject.content_rule_enable_fips_mode

great, I will do that, we were also depending on go_templating to be able to fetch different kube API, which is defined in warning part of the rule in the compliance operator dynamically based on value of the xccdf variables, among other things

**Comment by Honny1** [comment]
@Vincent056 Okay

**Review by Honny1 — APPROVED** [review]

**Review by yuumasato — APPROVED** [review]
Thank you for the review @Honny1

/lgtm

**Comment by theboringstuff** [comment]
This will not be included in `v0.1.73`? Is there WA for this issue? I encounter similar problem for openshift cluster
```
Fetching URI: '/api/v1/namespaces/-/pods?labelSelector=app%3Dkube-controller-manager'
debug: Applying filter '[[.items[0].spec.containers[0].args[] | select(. | match("--port=[1-9]*[1-9]+") )] | length | if . == 0 then true else false end]' to path '/api/v1/namespaces/-/pods?labelSelector=app%3Dkube-controller-manager'
debug: Error while filtering: cannot iterate over: null
debug: Persisting warnings to output file
FATAL:Error fetching resources: couldn't filter '{"kind":"PodList","apiVersion":"v1","metadata":{"resourceVersion":"190509111"},"items":[]}
': cannot iterate over: null
```

**Comment by yuumasato** [comment]
As #11858 is going into 0.1.73, I think it makes sense to backport this fix there.
@ComplianceAsCode/red-hatters do you guys concur?

**Pull request merged**